main.go

```diff
@@ -38,36 +42,81 @@ func avsSyncMain(cliCtx *cli.Context) error {
-		return err
-	}
-	ecdsaAddr := crypto.PubkeyToAddress(ecdsaPrivKey.PublicKey)
-	logger.Debug("ECDSA Address", "address", ecdsaAddr.Hex())
+	writerTimeout := cliCtx.Duration(WriterTimeoutDurationFlag.Name)
+	readerTimeout := cliCtx.Duration(ReaderTimeoutDurationFlag.Name)
 
 	ethHttpClient, err := eth.NewClient(cliCtx.String(EthHttpUrlFlag.Name))
 	if err != nil {
 		logger.Fatalf("Cannot create eth client", "err", err)
 	}
 
-	rpcCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	rpcCtx, cancel := context.WithTimeout(context.Background(), readerTimeout)
 	defer cancel()
 	chainid, err := ethHttpClient.ChainID(rpcCtx)
 	if err != nil {
 		logger.Fatalf("Cannot get chain id", "err", err)
 	}
-	// confusing interface, see https://github.com/Layr-Labs/eigensdk-go/issues/90
-	signerFn, _, err := signerv2.SignerFromConfig(signerv2.Config{
-		PrivateKey: ecdsaPrivKey,
-	}, chainid)
-	if err != nil {
-		logger.Errorf("Cannot create signer", "err", err)
-		return err
+
+	var wallet walletsdk.Wallet
+	operatorEcdsaPrivKeyHexStr := cliCtx.String(EcdsaPrivateKeyFlag.Name)
+	if len(operatorEcdsaPrivKeyHexStr) > 0 {
+		ecdsaPrivKey, err := crypto.HexToECDSA(operatorEcdsaPrivKeyHexStr)
+		if err != nil {
+			return fmt.Errorf("Cannot create ecdsa private key: %w", err)
+		}
+
+		signerV2, address, err := signerv2.SignerFromConfig(signerv2.Config{PrivateKey: ecdsaPrivKey}, chainid)
+		if err != nil {
+			return err
+		}
+		wallet, err = walletsdk.NewPrivateKeyWallet(ethHttpClient, signerV2, address, logger)
+		if err != nil {
+			return err
+		}
+	} else {
+		fbAPIKey := cliCtx.String(FireblocksAPIKeyFlag.Name)
+		fbSecretPath := cliCtx.String(FireblocksAPISecretPathFlag.Name)
+		fbBaseURL := cliCtx.String(FireblocksBaseURLFlag.Name)
+		fbVaultAccountName := cliCtx.String(FireblocksVaultAccountNameFlag.Name)
+		if fbAPIKey == "" {
+			return errors.New("Fireblocks API key is not set")
+		}
+		if fbSecretPath == "" {
+			return errors.New("Fireblocks API secret is not set")
+		}
+		if fbBaseURL == "" {
+			return errors.New("Fireblocks base URL is not set")
+		}
+		if fbVaultAccountName == "" {
+			return errors.New("Fireblocks vault account name is not set")
+		}
+
+		secretKey, err := os.ReadFile(fbSecretPath)
+		if err != nil {
+			return fmt.Errorf("Cannot read fireblocks secret from %s: %w", fbSecretPath, err)
+		}
+		fireblocksClient, err := fireblocks.NewClient(
+			fbAPIKey,
+			[]byte(secretKey),
+			fbBaseURL,
+			writerTimeout,
+			logger,
+		)
+		if err != nil {
+			return err
+		}
+		wallet, err = walletsdk.NewFireblocksWallet(fireblocksClient, ethHttpClient, fbVaultAccountName, logger)
+		if err != nil {
+			return err
+		}
 	}
 
-	txMgr := txmgr.NewSimpleTxManager(ethHttpClient, logger, signerFn, ecdsaAddr)
+	sender, err := wallet.SenderAddress(context.Background())
+	if err != nil {
+		return fmt.Errorf("Cannot get sender address: %w", err)
+	}
+	logger.Infof("Sender address: %s", sender.Hex())
```

Comment:
Is this needed, or was this only used for debugging?
Or should this be at the debug level?

Reply:
I left it at info level because it seemed useful enough and it's logged only once at initialization